### PR TITLE
infra: move jobs out of travis

### PR DESCRIPTION
**Pull request by nick-mancuso**
Attempting to fix failure noted at https://app.travis-ci.com/github/checkstyle/checkstyle/jobs/577739286 and https://github.com/checkstyle/checkstyle/pull/11990/checks?check_run_id=7513176563

### CODE REVIEW

**Comment by romani** [comment]
Even it failing due to other problem, but at least spell check is working, so let's merge this.

**Comment by rnveach** [comment]
Are you sure we can remove installation of xmlstarlet as it is used as part of some shells/CI?

**Comment by nick-mancuso** [comment]
I think it is best to move all remaining builds from Travis, it has proven unreliable time and time again. We can revisit Travis later if they improve free model.

File: .travis.yml.bak [hunk 1]
```diff
@@ -29,13 +29,13 @@ jobs:
...
-        - CMD="mvn -e --no-transfer-progress clean integration-test failsafe:verify
-          -DargLine='-Xms1024m -Xmx2048m'"
-        - DEPLOY="true"
-        - USE_MAVEN_REPO="true"
+#    - jdk: openjdk11
+#      env:
+#        - DESC="tests and deploy"
+#        - CMD="mvn -e --no-transfer-progress clean integration-test failsafe:verify
+#          -DargLine='-Xms1024m -Xmx2048m'"
+#        - DEPLOY="true"
```
Comment (romani) [inline .travis.yml.bak:37]: This job is fancy.
This variable trigger post build execution  to deploy to maven snapshot public repository, by secret credentials.
Better to keep it Travis.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .travis.yml.bak (hunk 1 above)
Comment (romani) [reply .travis.yml.bak:37]: It is ok to disable or keep it failing.

Comment (nick-mancuso) [reply .travis.yml.bak]: Done, it will be the only job left in Travis now.

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -8,6 +8,14 @@ auto_cancel:
   running:
     when: "branch != 'master'"
 blocks:
+  - name: "Shell Check"
+    task:
+      jobs:
+      - name: Execute shellcheck
+        commands:
+          - checkout
+          - shellcheck ./.ci/*.sh && shellcheck release.sh
```
Comment (romani) [inline .semaphore/semaphore.yml:17]: This update put others execution in sequence after this.
Look at graph of semaphore - weird.
Let's keep spellcheck in Travis it looks ok in execution.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .semaphore/semaphore.yml (hunk 2 above)
Comment (nick-mancuso) [reply .semaphore/semaphore.yml]: I made a new github workflow for shellcheck. We have an open issue and ongoing work for shellcheck suppressions, I do not want to block PRs and worry about re-running CI (or local execution) to verify shellcheck execution succeeds.

**Comment by nick-mancuso** [comment]
shellcheck: https://github.com/checkstyle/checkstyle/runs/7533823052?check_suite_focus=true

no-error-hibernate-search: https://circleci.com/gh/checkstyle/checkstyle/162557?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Pull request merged**